snappy_wrappers/wrappers/somatic_variants_checking/summarize-vcf.py

```diff
@@ -0,0 +1,254 @@
+    minimal=1,
+    limited=5,
+    bed_file="",
+    hard_file="",
+    filter_file=False,
+    padding=0,
+):
+    infor = {
+        "total_v_count": 0,
+        "v_inside_exom": 0,
+        "v_outside_exom": 0,
+        "v_in_hard_regions": 0,
+        "n_snps": 0,
+        "n_indels": 0,
+        "minimal_rp_snvs_exom": 0,
+        "minimal_rp_snvs_nexom": 0,
+        "limited_rp_snvs_exom": 0,
+        "limited_rp_snvs_nexom": 0,
+        "strong_rp_snvs_exom": 0,
+        "strong_rp_snvs_nexom": 0,
+        "indels_length": [],
+        "mt_classes": [0] * 6,
+        "vaf": [],
+    }
+    if filter_file:
+        for variant in vcf_file:
+            infor["total_v_count"] += 1
+            # Gathering information of variants in comparison to interested regions
+            if variant.CHROM in contigs:
+                if len(variant.ALT) > 1:
+                    raise ValueError(
+                        "Multiallelic variants are not supported"
+                    )  # checking for multiallelic
+                infor["vaf"].append(float(variant.format("AF")[1][0]))
+                # Counting variants in or out of exom and variants with different support reads
+                if check_variant_in_bed(
+                    variant.CHROM, variant.start, variant.end, bed_file, padding=0
+                ):
+                    infor["v_inside_exom"] += 1
+                    infor[str(check_sp_read(variant, minimal, limited)) + "_rp_snvs_exom"] += 1
+                else:
+                    infor["v_outside_exom"] += 1
+                    infor[str(check_sp_read(variant, minimal, limited)) + "_rp_snvs_nexom"] += 1
+
+                # Need to check multi allelic. Users shouldn't input multi allelic vcf file.
+                if get_variant_type(variant.REF, variant.ALT[0]) == "snp":
+                    infor["n_snps"] += 1
+                    infor["mt_classes"] = assign_class_snvs(variant, infor["mt_classes"])
+                else:
+                    # More for indels
+                    infor["n_indels"] += 1
+                    infor["indels_length"].append(abs(len(variant.REF) - len(variant.ALT[0])))
+            # Gathering information of variants in comparison to hard mapped regions
+
+            if variant.CHROM in hard_contigs:
+                if check_variant_in_bed(
+                    variant.CHROM, variant.start, variant.end, hard_file, padding=0
+                ):
+                    infor["v_in_hard_regions"] += 1
+        return infor
+    else:
+        total_v_count = 0
+        for variant in vcf_file:
+            total_v_count += 1
+        return total_v_count
+
+
+def main():
+    classes = ["T>G", "T>C", "T>A", "C>G", "C>T", "C>A"]
+    # reading input
+    full_vcf = VCF(args.rawvcf)
+    filter_vcf = VCF(args.filtered_vcf)
+    sample = filter_vcf.samples[1]
+    ######################
+    bed_intervals = tabix.open(args.exom_bedfile)
+    contigs = get_contigs_from_bed_file(args.exom_bedfile)
```

Comment:
I don't understand why you need the contigs.

Reply:
tabix will give an error if the bed file doesn't contain the contig that tabix queries.